src/components/plots/GeneReorderTool.jsx

```diff
@@ -0,0 +1,74 @@
+import React, { useEffect, useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import PropTypes from 'prop-types';
+
+import { arrayMoveImmutable } from 'utils/array-move';
+import { updatePlotConfig } from 'redux/actions/componentConfig';
+import { loadGeneExpression } from 'redux/actions/genes';
+import HierarchicalTreeGenes from 'components/plots/hierarchical-tree-genes/HierarchicalTreeGenes';
+
+const GeneReorderTool = (props) => {
+  const { plotUuid } = (props);
+
+  const dispatch = useDispatch();
+
+  const config = useSelector((state) => state.componentConfig[plotUuid]?.config);
+
+  const experimentId = useSelector((state) => state.componentConfig[plotUuid]?.experimentId);
+
+  const loadedMarkerGenes = useSelector(
+    (state) => state.genes.expression.views[plotUuid]?.data,
+  ) || [];
```

Comment:
Two things:
- In the api we can't but in the ui we have access to the null-coalescing operator (`??`) which is better than `||` because it only works with `null` or `undefined`, so it's more specific.

- In `composeGeneTree` you are handling the case where `treeGenes` is `undefined`/`null`, so I think you don't really need to handle it in the `useSelector` as well, just one of these two places should be ok (since `composeGeneTree` has to deal with more cases maybe it's better to keep the handling there).